crates/bevy_core/src/time/time.rs

```diff
@@ -33,115 +56,411 @@ impl Time {
 
     pub(crate) fn update_with_instant(&mut self, instant: Instant) {
         if let Some(last_update) = self.last_update {
-            self.delta = instant - last_update;
-            self.delta_seconds_f64 = self.delta.as_secs_f64();
+            self.raw_delta = instant - last_update;
+            self.raw_delta_seconds = self.raw_delta.as_secs_f32();
+            self.raw_delta_seconds_f64 = self.raw_delta.as_secs_f64();
+
+            if self.relative_speed != 1.0 {
+                // just to be safe
+                self.delta = self.raw_delta.mul_f64(self.relative_speed_f64());
+            } else {
+                self.delta = self.raw_delta;
+            }
+
             self.delta_seconds = self.delta.as_secs_f32();
+            self.delta_seconds_f64 = self.delta.as_secs_f64();
+            self.elapsed_since_startup += self.delta;
+        } else {
+            self.first_update = Some(instant);
+
+            if self.relative_speed != 1.0 {
+                // just to be safe
+                self.elapsed_since_startup =
+                    (instant - self.startup).mul_f64(self.relative_speed as f64);
```

Comment:
Why not use `self.relative_speed_f64()` here like above? Assuming you don't remove it of course.

Reply:
Good point.